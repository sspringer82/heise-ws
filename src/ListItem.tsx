import React from 'react';
import Person from './Person';

type Props = {
  person: Person;
  onDelete: (id: number) => void;
};

const ListItem: React.FC<Props> = ({ person, onDelete }) => {
  return (
    <tr key={person.id}>
      <td>{person.id}</td>
      <td>{person.firstName}</td>
      <td>{person.lastName}</td>
      <td>{person.birthdate}</td>
      <td>{person.street}</td>
      <td>{person.city}</td>
      <td>{person.zipCode}</td>
      <td>
        <button onClick={() => onDelete(person.id)}>l√∂schen</button>
      </td>
    </tr>
  );
};

export default ListItem;
