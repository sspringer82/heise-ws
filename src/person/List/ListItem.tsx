import { IconButton, TableCell, TableRow } from '@mui/material';
import React from 'react';
import Person from '../Person';
import DeleteIcon from '@mui/icons-material/Delete';
import { Link } from 'react-router-dom';

type Props = {
  person: Person;
  onDelete: (id: number) => void;
};

const ListItem: React.FC<Props> = ({ person, onDelete }) => {
  return (
    <TableRow key={person.id}>
      <TableCell data-testid="id">{person.id}</TableCell>
      <TableCell data-testid="firstName">{person.firstName}</TableCell>
      <TableCell data-testid="lastName">{person.lastName}</TableCell>
      <TableCell data-testid="birthdate">{person.birthdate}</TableCell>
      <TableCell data-testid="street">{person.street}</TableCell>
      <TableCell data-testid="city">{person.city}</TableCell>
      <TableCell data-testid="zipCode">{person.zipCode}</TableCell>
      <TableCell>
        <IconButton
          aria-label="delete"
          color="primary"
          onClick={() => onDelete(person.id)}
          data-testid="delete-btn"
        >
          <DeleteIcon />
        </IconButton>
      </TableCell>
      <TableCell>
        <Link to={`/edit/${person.id}`}>bearbeiten</Link>
      </TableCell>
    </TableRow>
  );
};

export default ListItem;
