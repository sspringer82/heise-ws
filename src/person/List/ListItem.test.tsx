import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ListItem from './ListItem';

const person = {
  id: 444,
  firstName: 'Dorothea',
  lastName: 'Hammes',
  birthdate: '1955-11-18T14:22:15.431Z',
  street: '385 Conn Overpass Apt. 740',
  city: 'Hackensack',
  zipCode: '98692-7937',
};

describe('ListItem', () => {
  it('snapshot test', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <table>
          <tbody>
            <ListItem person={person} onDelete={jest.fn()} />
          </tbody>
        </table>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly', () => {
    render(
      <MemoryRouter>
        <table>
          <tbody>
            <ListItem person={person} onDelete={jest.fn()} />
          </tbody>
        </table>
      </MemoryRouter>
    );

    expect(screen.getByTestId('firstName')).toHaveTextContent('Dorothea');
    expect(screen.getByTestId('lastName')).toHaveTextContent('Hammes');
  });

  it('should delete an entry', () => {
    const onDelete = jest.fn();

    render(
      <MemoryRouter>
        <table>
          <tbody>
            <ListItem person={person} onDelete={onDelete} />
          </tbody>
        </table>
      </MemoryRouter>
    );

    fireEvent.click(screen.getByTestId('delete-btn'));

    expect(onDelete).toHaveBeenCalled();
    expect(onDelete).toHaveBeenCalledTimes(1);
    expect(onDelete).toHaveBeenCalledWith(444);
  });
});
