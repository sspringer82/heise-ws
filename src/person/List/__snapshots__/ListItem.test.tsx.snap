// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem snapshot test 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
          data-testid="id"
        >
          444
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
          data-testid="firstName"
        >
          Dorothea
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
          data-testid="lastName"
        >
          Hammes
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
          data-testid="birthdate"
        >
          1955-11-18T14:22:15.431Z
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
          data-testid="street"
        >
          385 Conn Overpass Apt. 740
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
          data-testid="city"
        >
          Hackensack
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
          data-testid="zipCode"
        >
          98692-7937
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
        >
          <button
            aria-label="delete"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
            data-testid="delete-btn"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
        >
          <a
            href="/edit/444"
          >
            bearbeiten
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
